File: flutter/parent_app/lib/features/journey_history/repository/models/journey_history_entry.dart
import 'package:flutter/foundation.dart';

import '../../../../core/domain.dart';

/// One past leg — a single morning or afternoon journey on a single day.
///
/// P-05's durable record: "a parent who missed what happened must be able to scroll back
/// and find it" (docs/05-ui/PARENT_APP.md, said of P-08 and equally true here).
@immutable
class JourneyHistoryEntry {
  const JourneyHistoryEntry({
    required this.serviceDate,
    required this.direction,
    required this.state,
    this.vehicleDisplayName,
    this.stopName,
    this.eventAt,
  });

  /// The school day this leg belongs to, in school-local terms — a calendar date, not an
  /// instant (docs/04-api/API_STANDARDS.md: dates are `YYYY-MM-DD`, distinct from
  /// timestamps).
  final DateTime serviceDate;

  final JourneyDirection direction;
  final JourneyState state;
  final String? vehicleDisplayName;
  final String? stopName;

  /// When the terminal event on this leg happened — boarding, arrival, or handover.
  final SchoolTime? eventAt;
}

File: backend/guardian-parent/src/main/java/com/guardian/parent/domain/JourneyHistoryEntry.java
package com.guardian.parent.domain;

import java.time.Instant;
import java.time.LocalDate;
import java.util.Objects;

/**
 * One past leg — a single morning or afternoon journey on a single day (P-05).
 *
 * <p>The durable record a parent scrolls back through. It is built from {@code boarding_events},
 * which is append-only and is the evidence, rather than from any summary table: a summary would be
 * a second version of what happened, and the two would eventually disagree.
 *
 * @param serviceDate the school day this leg belongs to. A {@link LocalDate}, not an instant — a
 *     school day is a calendar fact, and an instant would put the boundary in the wrong day for any
 *     school not on UTC (BR-CFG-006).
 * @param eventAt the terminal event on the leg, in UTC. Null where the leg produced no boarding
 *     record at all, which is itself the story on a no-show.
 */
public record JourneyHistoryEntry(
    LocalDate serviceDate,
    JourneyLeg.Direction direction,
    JourneyState state,
    String vehicleDisplayName,
    String stopName,
    Instant eventAt) {

  public JourneyHistoryEntry {
    Objects.requireNonNull(serviceDate, "serviceDate");
    Objects.requireNonNull(direction, "direction");
    Objects.requireNonNull(state, "state");
  }
}
